### PR TITLE
Feature/orb read policies | Unit test concurrency problem

### LINKED ISSUE #203: Policy Read

### LINKED ISSUE #206: Configure Policy Project

### PR DESCRIPTION
**Pull request by dscabral**
Revolves #214, resolves #203, resolves #206

### CODE REVIEW

**Comment by dscabral** [comment]
I'm so sorry for this big PR, but i had to change a lot of files to solve the concurrency problem in unit tests...
Some tests had interdependence and it was not ok. So I isolated the creation from the conflict scenario on creation unit tests. So now, even if the creation or conflict tests, run in a inverted order, they will throw a error.

File: cmd/fleet/main.go [hunk 1]
```diff
@@ -15,7 +15,9 @@ import (
 	mflog "github.com/mainflux/mainflux/logger"
 	mfnats "github.com/mainflux/mainflux/pkg/messaging/nats"
 	"github.com/ns1labs/orb/fleet"
-	"github.com/ns1labs/orb/fleet/api"
+	fleetgrpc "github.com/ns1labs/orb/fleet/api/grpc"
+	http2 "github.com/ns1labs/orb/fleet/api/http"
```
Comment (weyrick) [inline cmd/fleet/main.go:19]: let's call this `fleethttp` for clarity

File: cmd/policies/main.go [hunk 2]
```diff
@@ -58,8 +58,11 @@ func main() {
 	svcCfg := config.LoadBaseServiceConfig(envPrefix, httpPort)
 	dbCfg := config.LoadPostgresConfig(envPrefix, svcName)
 	jCfg := config.LoadJaegerConfig(envPrefix)
+	fleetGRPCCfg := config.LoadGRPCConfig("orb", "fleet")
 	policiesGRPCCfg := config.LoadGRPCConfig("orb", "policies")
 
+	fmt.Sprintf(fleetGRPCCfg.Port)
```
Comment (weyrick) [inline cmd/policies/main.go:64]: debug?

File: cmd/policies/main.go [hunk 3]
```diff
@@ -207,16 +210,16 @@ func connectToGRPC(cfg config.GRPCConfig, logger *zap.Logger) *grpc.ClientConn {
...
 }
 
-func startHTTPServer(svc policies.Service, cfg config.BaseSvcConfig, logger *zap.Logger, errs chan error) {
+func startHTTPServer(tracer opentracing.Tracer, svc policies.Service, cfg config.BaseSvcConfig, logger *zap.Logger, errs chan error) {
 	p := fmt.Sprintf(":%s", cfg.HttpPort)
 	if cfg.HttpServerCert != "" || cfg.HttpServerKey != "" {
 		logger.Info(fmt.Sprintf("Policies service started using https on port %s with cert %s key %s",
 			cfg.HttpPort, cfg.HttpServerCert, cfg.HttpServerKey))
-		errs <- http.ListenAndServeTLS(p, cfg.HttpServerCert, cfg.HttpServerKey, http2.MakeHandler(svcName, svc))
+		errs <- http.ListenAndServeTLS(p, cfg.HttpServerCert, cfg.HttpServerKey, http2.MakeHandler(tracer, svcName, svc))
```
Comment (weyrick) [inline cmd/policies/main.go:218]: for consistency why don't we go ahead and also change this `http2` to `policieshttp`

File: fleet/pb/fleet.proto [hunk 4]
```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
```
Comment (weyrick) [inline fleet/pb/fleet.proto:1]: did you add this to the `proto` target in Makefile? policies is already there. I'm not sure why there is both a generated `fleet.pb.go` and `fleet_grpc.pb.go`, is one old maybe?

File: fleet/api/grpc/server.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	return &grpcServer{
+		retrieveAgent: kitgrpc.NewServer(
+			kitot.TraceServer(tracer, "retrieve_agent")(retrieveAgentEndpoint(svc)),
+			decodeRetrieveAgentRequest,
+			encodeAgentResponse,
+		),
+	}
+}
+
+func (gs *grpcServer) RetrievePolicy(ctx context.Context, req *pb.AgentByIDReq) (*pb.AgentRes, error) {
```
Comment (weyrick) [inline fleet/api/grpc/server.go:31]: says `RetrievePolicy` - copy/paste error?

File: fleet/api/grpc/endpoint.go [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"context"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/ns1labs/orb/fleet"
+)
+
+func retrieveAgentEndpoint(svc fleet.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		return nil, nil
```
Comment (weyrick) [inline fleet/api/grpc/endpoint.go:11]: is this non functional for now?

File: policies/policies.go [hunk 7]
```diff
@@ -33,6 +33,34 @@ type Dataset struct {
...
+	// CreatePolicy creates new agent Policy
+	AddPolicy(ctx context.Context, token string, p Policy, format string, policyData string) (Policy, error)
+
+	// RetrievePolicyByID retrieving policy by id with token
+	ViewPolicyByID(ctx context.Context, token string, policyID string) (Policy, error)
+
+	// RetrieveAll
+	ListPolicies(ctx context.Context, token string, pm PageMetadata) (Page, error)
+
+	// RetrievePolicyByIDInternal gRPC version of retrieving policy by id with no token
```
Comment (weyrick) [inline policies/policies.go:51]: change comments to match method names in this block

**Review by weyrick — CHANGES_REQUESTED** [review]

File: cmd/fleet/main.go (hunk 1 above)
Comment (dscabral) [reply cmd/fleet/main.go:19]: Ok, nice tip.
Done

File: cmd/policies/main.go (hunk 2 above)
Comment (dscabral) [reply cmd/policies/main.go:64]: Yeap, sorry
Done

File: cmd/policies/main.go (hunk 3 above)
Comment (dscabral) [reply cmd/policies/main.go:218]: Done

File: fleet/api/grpc/server.go (hunk 5 above)
Comment (dscabral) [reply fleet/api/grpc/server.go:31]: Should be RetrieveAgent
Done

File: fleet/api/grpc/endpoint.go (hunk 6 above)
Comment (dscabral) [reply fleet/api/grpc/endpoint.go:11]: I'll do it in this PR, was my fault.

File: policies/policies.go (hunk 7 above)
Comment (dscabral) [reply policies/policies.go:51]: Done

**Review by weyrick — APPROVED** [review]

**Pull request merged**